WBooks/Views/Login/LoginViewController.swift

```diff
@@ -21,84 +21,58 @@ final class LoginViewController: UIViewController, GIDSignInUIDelegate {
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        //Sets the Login and LogOut buttons
-        _myView.loginButton.addTarget(self, action: #selector(loginClick(sender:)), for: .touchUpInside)
+        //Sets the LogIn and LogOut buttons
+        _myView.logInButton.addTarget(self, action: #selector(logInClick(sender:)), for: .touchUpInside)
         _myView.logOutButton.addTarget(self, action: #selector(logOutClick(sender:)), for: .touchUpInside)
-        _myView.statusText.textAlignment = NSTextAlignment.center
-         
+        
         GIDSignIn.sharedInstance().uiDelegate = self
+        GIDSignIn.sharedInstance().delegate = self
         
-        // Automatically sign in the user.
-        GIDSignIn.sharedInstance().signInSilently()
+        // Automatically signs in the user, if there is a user already authenticated.
+        if GIDSignIn.sharedInstance().hasAuthInKeychain() == true {
+            GIDSignIn.sharedInstance().signInSilently()
+        }
         
-        // [START_EXCLUDE]
-        NotificationCenter.default.addObserver(self,
-                                               selector: #selector(receiveToggleAuthUINotification(_:)),
-                                               name: NSNotification.Name(rawValue: "ToggleAuthUINotification"),
-                                               object: nil)
         _myView.statusText.text = "Initialized Swift app..."
         toggleAuthUI()
-        // [END_EXCLUDE]
     }
     
     @objc
-    func loginClick(sender: UIButton) {
-        print("Login")
+    func logInClick(sender: UIButton) {
+        _myView.statusText.text = "Logging..."
         GIDSignIn.sharedInstance().signIn()
     }
     
-    // [START signout_tapped]
     @objc
     func logOutClick(sender: AnyObject) {
         GIDSignIn.sharedInstance().signOut()
-        // [START_EXCLUDE silent]
         _myView.statusText.text = "Signed out."
         toggleAuthUI()
-        // [END_EXCLUDE]
     }
   
-    // [START toggle_auth]
     //Defines which button has be to shown, depending if the user has to log in, of if it's already logged.
     func toggleAuthUI() {
```

Comment:
Personally, I would prefer something like `changeUI(forLoggedIn loggedIn: Bool)`, so this function doesn't need to be coupled to GIDSignIn. Whenever you call this function you already know if you have a user signed in or not.